core/src/main/scala/zio/config/AnnotatedRead.scala

```diff
@@ -18,5 +18,6 @@ object AnnotatedRead {
   sealed trait Annotation
   object Annotation {
     case object NonDefaultValue extends Annotation
+    case object Recursive       extends Annotation
```

Comment:
Not required anymore

Reply:
With making `lazy val sources` working now, remove `Recursive` annotations. This was part of a hacky solution as part of testing